pkg/vault/contracts/Vault.sol

```diff
@@ -740,7 +740,7 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
             poolData.updateRawAndLiveBalance(
                 i,
                 poolData.balancesRaw[i] + amountInRaw - locals.totalFeesRaw,
-                Rounding.ROUND_UP
+                Rounding.ROUND_DOWN
```

Comment:
Should we just always round down when computing live balances? This is currently rounded up in only one place: tokenIn on a swap (Vault.sol: L461). Raw amounts were already calculated rounding up.

If so, we could just remove the rounding argument and save some bytes/gas.

This rounding has no effect on the token transfers; those use the raw values. It could only have an impact on future yield fees, which use the lastLive balances.

Reply:
> This rounding has no effect on the token transfers; those use the raw values. It could only have an impact on future yield fees, which use the lastLive balances.

Exactly. The thing is that this should only be used for the next operation; we've already done the main hook math to get raw results at this point.

> Should we just always round down when computing live balances? This is currently rounded up in only one place: tokenIn on a swap (Vault.sol: L461). Raw amounts were already calculated rounding up.

That should probably be the case yes. I think I'll open a PR just with these two changes.

Reply:
> If so, we could just remove the rounding argument and save some bytes/gas.

My understanding is that there's one case where it does make sense to round up: in `addLiquidity` before the main hook. Rounding up there helps to compute less BPT out, as stated in the comments: https://github.com/balancer/balancer-v3-monorepo/blob/fb2753a76cf3cb650c2adf412735d84d931b67af/pkg/vault/contracts/Vault.sol#L518-L531.

If that's the case, we can't really remove the direction from the function.

Reply:
I've removed this change from here in favor of #947